### PR TITLE
fix: check for empty array

### LINKED ISSUE #3: Error: TypeError: Cannot read property 'map' of undefined
In the post step, getting the following:

```
Post job cleanup.
Found server pid: 1643
Killing server pid: 1643
Error: TypeError: Cannot read property 'map' of undefined
```

Sorry, doesn't provide any more info than that.

---

screenshot-20220208T102136-SQSGOUx6

### PR DESCRIPTION
**Pull request by moltar**
Not sure about the transpiled code. I used `npm run build`, but it seems to have produced a lot of change. Not sure why.

Closes #3

### CODE REVIEW

File: src/utils/uploadArtifacts.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { cacheDir } from './constants';
 
 export async function uploadArtifacts() {
   const list = await artifactApi.listArtifacts();
-  const existingArtifacts = list.artifacts.map((artifact) => artifact.name);
+  const existingArtifacts = list.artifacts?.map((artifact) => artifact.name) || []
```
Comment (moltar) [inline src/utils/uploadArtifacts.ts:10]: This is the only change I've made.

Comment (felixmosh) [reply src/utils/uploadArtifacts.ts:10]: Cool, thank you.

**Comment by felixmosh** [comment]
This is weird, how the list is `undefined`?

Can you check what the API return at this endpoint?
https://api.github.com/repos/{{user}}/{{repo}}/actions/artifacts

**Comment by felixmosh** [comment]
I've released this fix.

**Comment by moltar** [comment]
> This is weird, how the list is undefined?

Yeah, pretty strange, no idea 🤷🏼‍♂️ 

```
❯ gh api repos/owner/repo/actions/artifacts
```

```json
{
  "total_count": 83,
  "artifacts": [
    {
      "id": 1597810361,
      "node_id": "MDg6QXJ0aWZhY3QxXYZk3ODEwMzY=",
      "name": "fdbe1fd5xyfa61cb91",
      "size_in_bytes": 2497,
      "url": "https://api.github.com/repos/owner/repo/actions/artifacts/1597810361",
      "archive_download_url": "https://api.github.com/repos/owner/repo/actions/artifacts/1597810361/zip",
      "expired": false,
      "created_at": "2022-02-08T10:11:02Z",
      "updated_at": "2022-02-08T10:11:03Z",
      "expires_at": "2022-05-09T10:08:41Z"
    },
    {
       // ...
    }
  ]
```

**Pull request merged**